src/Accounts/Authentication/Utilities/AzConfigReader.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Shared.Config;
+using Microsoft.Azure.PowerShell.Common.Config;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Utilities
+{
+    static public class AzConfigReader
+    {
+
+        public static IAzureSession Session
+        {
+            get
+            {
+                return AzureSession.Instance;
+            }
+        }
+
+        static public bool IsWamEnabled()
+        {
+            if (!Session.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var config))
```

Comment:
Session can be null?

Reply:
AzureSession is initialized when Az.Accounts is imported and so is not expected to be null. If null an exception is expected.